### PR TITLE
Turn on nullable semantic analysis by default

### PR DESCRIPTION
**Pull request by 333fred**
Commit 1 ties the nullable semantic analysis to whether or not we will run the nullable walker during compilation.
Commit 2 fixes some DebugVerifier asserts caught after the verifier is run over more code.
Commit 3 updates the method compiler to not use the rewritten method body for emit.
Commit 4 adjusts IOperation to correctly retrieve anonymous properties, and adds a test. Caught by the debug bootstrap compiler.
Commit 5 adjust the BoundAnonymousObjectCreation handling and fixes https://github.com/dotnet/roslyn/issues/35044, thanks @cston for pointing out the correct way to do the comparisons.

Fixes https://github.com/dotnet/roslyn/issues/35374.

### CODE REVIEW

**Comment by 333fred** [comment]
@jasonmalinowski @chsienki @dotnet/roslyn-compiler for review please.

File: src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs [hunk 1]
```diff
@@ -182,7 +182,19 @@ internal override CommonAnonymousTypeManager CommonAnonymousTypeManager
         // Note that this is intentionally not conditioned off RunNullableWalker currently. Once we fully enable
```
Comment (jasonmalinowski) [inline src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs]: Does this comment need updating?

File: src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs [hunk 2]
```diff
@@ -182,7 +182,19 @@ internal override CommonAnonymousTypeManager CommonAnonymousTypeManager
...
         // the nullable analysis semantic model feature and feel confident in it, we only run the analysis if
         // the feature flag is set.
-        internal bool NullableSemanticAnalysisEnabled => Feature("run-nullable-analysis") is "true";
+        internal bool NullableSemanticAnalysisEnabled
+        {
+            get
+            {
+                const string NullableAnalysisFlag = "run-nullable-analysis";
+                if (Feature(NullableAnalysisFlag) is "false")
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs]: >Feature(NullableAnalysisFlag) [](start = 20, length = 29)

Please store in a local.

At that point, the constant above can be inlined.

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs [hunk 3]
```diff
@@ -435,6 +434,28 @@ internal static ImmutableArray<BoundNode> CreateInvalidChildrenFromArgumentsExpr
...
+                    return null;
+                }
+
+                var currentDeclaration = declarations[currentDeclarationIndex];
+
+                // https://github.com/dotnet/roslyn/issues/35044: This works for simple success cases, but does not work for failures. Likely will have to do something more complicated here involving rebinding the
+                // declarators based on the newly constructed anonymous type symbol above and matching them to the existing symbol
+                if ((object)currentProperty == currentDeclaration.Property ||
+                    (currentDeclaration.Property.Name == currentProperty.Name &&
+                     currentDeclaration.Property.Type.Equals(currentProperty.Type, TypeCompareKind.ConsiderEverything | TypeCompareKind.AllNullableIgnoreOptions)))
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs]: `currentProperty.MemberIndexOpt == currentDeclaration.MemberIndexOpt`

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs [hunk 4]
```diff
@@ -1085,6 +1085,36 @@ void verifySpeculativeTypeInfo(int position, PublicNullableFlowState expectedFlo
                 specTypeInfo = model.GetSpeculativeTypeInfo(position, newCoalesce, SpeculativeBindingOption.BindAsExpression);
                 Assert.Equal(PublicNullableFlowState.NotNull, specTypeInfo.Nullability.FlowState);
             }
+
+        }
```
Comment (cston) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs]: Unnecessary blank line.

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 5]
```diff
@@ -38970,6 +38970,27 @@ public void AnonymousTypes_16()
...
+        [Fact]
+        [WorkItem(35044, "https://github.com/dotnet/roslyn/issues/35044")]
+        public void AnonymousTypes_17()
+        {
+            var source = @"
+class Program
+{
+    static void M(object o1, object? o2)
+    {
+        new { O1/*T:object!*/ = o1, O1/*T:object?*/ = o2 }.O1.ToString();;
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >; [](start = 73, length = 1)

Extra `;`.

**Pull request merged**